src/Sulu/Bundle/ContentBundle/EventListener/WebspaceSerializeEventSubscriber.php

```diff
@@ -69,5 +69,13 @@ public function onPostSerialize(ObjectEvent $event)
 
         $portalInformation = $event->getContext()->accept(array_values($portalInformation));
         $visitor->addData('portalInformation', $portalInformation);
+
+        $urls = [];
+        foreach ($webspace->getPortals() as $portal) {
+            $environment = $portal->getEnvironment($this->environment);
+            $urls = array_merge($urls, $environment->getUrls());
+        }
+        $urls = $event->getContext()->accept($urls);
+        $visitor->addData('urls', $urls);
```

Comment:
Why don't you add a `getUrls` method to the webspace, and use a virtual property? Then this functionality would also be available somewhere else, in case we need it.